### PR TITLE
patch vote added and tests passing

### PR DESCRIPTION
**Pull request by jjcrl**
Should the article returned a comment count ?

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -135,6 +135,62 @@ describe("ENDPOINT TESTING", () => {
...
+        const vote = { inc_votes: newVote };
+        return request(app)
+          .patch("/api/articles/1")
+          .send(vote)
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe('Invalid Input')
+          });
+      });
+    });
```
Comment (dav3rid) [inline __tests__/app.test.js:215]: May seem repetitive - but I'd still add tests for the article id being non-existent/invalid too

File: __tests__/app.test.js [hunk 2]
```diff
@@ -179,6 +179,28 @@ describe("ENDPOINT TESTING", () => {
...
       });
+      test("404: Should return 404 error when article_id does not exist", () => {
+        const newVote = 1;
+        const vote = { inc_votes: newVote };
+        return request(app)
+          .patch("/api/articles/1000")
+          .send(vote)
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe("Page Not Found");
```
Comment (dav3rid) [inline __tests__/app.test.js]: I think "Page not found" is a little misleading as to what may have gone wrong

File: db/controllers/articles.controller.js [hunk 3]
```diff
@@ -38,8 +38,11 @@ exports.updateArticleVote = (req, res, next) => {
 
   patchArticleVote(articles_id, inc_votes)
     .then((article) => {
-      res.status(201).send({article})
+      if (!article) {
+        return Promise.reject({ status: 404, msg: "Page Not Found" });
```
Comment (dav3rid) [inline db/controllers/articles.controller.js]: Something like "article not found" or "resource not found" could go here instead

File: __tests__/app.test.js (hunk 2 above)
Comment (jjcrl) [reply __tests__/app.test.js]: - [ X ] changed to Resource not found

File: db/controllers/articles.controller.js (hunk 3 above)
Comment (jjcrl) [reply db/controllers/articles.controller.js]: - [ X ]  as above

**Pull request merged**